include/seqan3/alignment/configuration/align_config_band.hpp

```diff
@@ -14,51 +14,104 @@
+ * \ingroup alignment_configuration
+ */
+struct lower_diagonal : public seqan3::detail::strong_type<int32_t, lower_diagonal>
+{
+    //!\brief The type of the strong type base class.
+    using base_t = seqan3::detail::strong_type<int32_t, lower_diagonal>;
+    // Import the base class constructors
+    using base_t::base_t;
+};
+
+/*!\brief A strong type representing the upper diagonal of the seqan3::align_cfg::band_fixed_size.
+ * \ingroup alignment_configuration
+ */
+struct upper_diagonal : public seqan3::detail::strong_type<int32_t, upper_diagonal>
+{
+    //!\brief The type of the strong type base class.
+    using base_t = seqan3::detail::strong_type<int32_t, upper_diagonal>;
+    // Import the base class constructors
+    using base_t::base_t;
+};
+
+/*!\brief Configuration element for setting a fixed size band.
  * \ingroup alignment_configuration
- *
- * \tparam band_t The type of the band.
  *
  * \details
  *
- * Configures the banded alignment algorithm. Currently only seqan3::static_band is allowed as argument.
- * If no band is configured for the alignment algorithm the full alignment matrix will be computed.
- * Before executing the algorithm the band is tested for valid settings, e.g. that the upper bound is not smaller than
- * the lower bound, or the band is not shifted out of the alignment matrix. If an invalid setting is detected, a
- * seqan3::invalid_alignment_configuration exception will be thrown.
+ * Configures the banded alignment algorithm. Currently only a fixed size band is allowed.
+ * The band is given in form of a seqan3::align_cfg::lower_diagonal and a seqan3::align_cfg::upper_diagonal.
+ * A diagonal represents the cells in the alignment matrix that are not crossed by the alignment either downwards by
+ * the lower diagonal or rightwards by the upper diagonal. Thus any computed alignment will be inside the area defined
+ * by the lower and the upper diagonal.
+ *
+ * If this configuration is default constructed or not set during the algorithm configuration the the full alignment
+ * matrix will be computed.
+ *
+ * During the construction of this configuration element or before the execution of the alignment algorithm the
+ * band configuration is validated. If the user provided an invalid band, e.g. the upper diagonal is smaller than
+ * the lower bound, or the alignment matrix would be ill configured such that the requested alignment method cannot
+ * be computed (because the global alignment requires the first cell and the last cell of the matrix to be reachable),
+ * then a seqan3::invalid_alignment_configuration will be thrown.
  *
  * ### Example
  *
  * \include test/snippet/alignment/configuration/align_cfg_band_example.cpp
  */
-template <typename band_t>
-//!\cond
-    requires std::same_as<band_t, static_band>
-//!\endcond
-struct band : public pipeable_config_element<band<band_t>, band_t>
+class band_fixed_size : public pipeable_config_element<band_fixed_size, seqan3::detail::empty_type>
```

Comment:
```suggestion
class band_fixed_size : public pipeable_config_element<band_fixed_size, seqan3::detail::empty_type>
```

Is this blocked by #1871?

Reply:
#1871 is merged :D

Reply:
I know